wapitiCore/attack/mod_drupal_enum.py

```diff
@@ -0,0 +1,117 @@
+import re
+import json
+import hashlib
+from os.path import join as path_join
+
+from wapitiCore.net.web import Request
+from wapitiCore.attack.attack import Attack
+from wapitiCore.language.vulnerability import LOW_LEVEL, _
+from wapitiCore.definitions.fingerprint import NAME as TECHNO_DETECTED
+
+MSG_TECHNO_VERSIONED = _("{0} {1} detected")
+MSG_NO_DRUPAL = _("No Drupal Detected")
+
+class mod_drupal_enum(Attack):
+    """Detect Drupal version."""
+    name = "drupal_enum"
+    PAYLOADS_HASH = "drupal_hash_files.json"
+    PAYLOADS_FILE_THEMES = "wordpress_themes.txt"
+
+    versions = []
+    def __init__(self, crawler, persister, logger, attack_options):
+        Attack.__init__(self, crawler, persister, logger, attack_options)
+        self.finished = False
+
+    def get_hash(self):
+        with open(path_join(self.DATA_DIR, self.PAYLOADS_HASH), errors="ignore") as hashes:
+            data = json.load(hashes)
+            return data
+
+    def detect_version(self, url):
+        vers = {}
+        data = self.get_hash()
+        for uri in data :
+            req = Request('{}{}'.format(url, uri))
+            rep = self.crawler.get(req)
+            if rep.status != 200:
+                continue
+
+            cont = rep.content.encode()
+            hash_content = hashlib.sha256(cont).hexdigest()
+
+            if uri in data :
+                if hash_content in data[uri] :
+                    vers[uri] = data[uri][hash_content]
+
+        i = 0
+        result =[]
```

Comment:
you can do
`self.versions = set.intersection(*[set(versions) for versions in vers.values()])`